File: Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression/simple_linear_regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sun May  7 15:41:13 2017

@author: akshay
"""

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

#importing data
dataset = pd.read_csv('Salary_Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:,1].values
                
#splitting data into training and test set
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 1/3, random_state = 0)


#Fitting simplelinear regression model to our data
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train, y_train)


# Predict the values
y_pred = regressor.predict(x_test)

#plot graph for training set
plt.scatter(x_train, y_train, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color = 'green')
plt.title('Salary vs Experience')
plt.xlabel('Years pf experience')
plt.ylabel('Salary')
plt.show()

#plot graph for the test set
plt.scatter(x_test, y_test, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color = 'green')
plt.title('Salary vs Experience')
plt.xlabel('Years pf experience')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression/simple _linear_regression.R
dataset = read.csv('Salary_Data.csv')

#splitting data into training and test sets
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


#fitting the Data set to the linear regressson model
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

#predict the test test results
predict(regressor, newdata = test_set)

#plot the results
#install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Savary vs Years of Experience') +
  xlab('Years of Experience') +
  ylab('Salary')

ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Savary vs Years of Experience') +
  xlab('Years of Experience') +
  ylab('Salary')